src/test/ui/const-generics/diagnostics.stderr

```diff
@@ -5,7 +5,16 @@ LL | struct A<const N: u8>;
    | ---------------------- similarly named struct `A` defined here
 LL | trait Foo {}
 LL | impl Foo for A<N> {}
-   |                ^ help: a struct with a similar name exists: `A`
+   |                ^
+   |
+help: a struct with a similar name exists
+   |
+LL | impl Foo for A<A> {}
+   |                ^
```

Comment:
We should likely filter cases like these, where we suggest the same type we are evaluating, but single letter structs should be uncommon enough that it isn't a problem in practice.